src/bodygraph.cpp

```diff
@@ -0,0 +1,577 @@
+#include "avatar.h"
+#include "bodygraph.h"
+#include "bodypart.h"
+#include "cursesdef.h"
+#include "damage.h"
+#include "generic_factory.h"
+#include "input.h"
+#include "ui_manager.h"
+
+#define BPGRAPH_MAXROWS 20
+#define BPGRAPH_MAXCOLS 40
+
+static const bodygraph_id bodygraph_full_body( "full_body" );
+
+namespace
+{
+
+generic_factory<bodygraph> bodygraph_factory( "bodygraph" );
+
+} // namespace
+
+/** @relates string_id */
+template<>
+const bodygraph &string_id<bodygraph>::obj() const
+{
+    return bodygraph_factory.obj( *this );
+}
+
+/** @relates string_id */
+template<>
+bool string_id<bodygraph>::is_valid() const
+{
+    return bodygraph_factory.is_valid( *this );
+}
+
+void bodygraph::load_bodygraphs( const JsonObject &jo, const std::string &src )
+{
+    bodygraph_factory.load( jo, src );
+}
+
+void bodygraph::reset()
+{
+    bodygraph_factory.reset();
+}
+
+const std::vector<bodygraph> &bodygraph::get_all()
+{
+    return bodygraph_factory.get_all();
+}
+
+void bodygraph::finalize_all()
+{
+    bodygraph_factory.finalize();
+}
+
+void bodygraph::check_all()
+{
+    bodygraph_factory.check();
+}
+
+void bodygraph::load( const JsonObject &jo, const std::string & )
+{
+    optional( jo, was_loaded, "parent_bodypart", parent_bp );
+    optional( jo, was_loaded, "fill_sym", fill_sym );
+    if( jo.has_string( "fill_color" ) ) {
+        fill_color = color_from_string( jo.get_string( "fill_color" ) );
+    }
+
+    if( !was_loaded || jo.has_array( "rows" ) ) {
```

Comment:
It doesn't look like you're setting `was_loaded` anywhere and UBSan complains about it too:
```gdb
../cataclysm-dda-git/src/bodygraph.cpp:69:10: runtime error: load of value 255, which is not a valid value for type 'bool'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../cataclysm-dda-git/src/bodygraph.cpp:69:10 in
```

Reply:
Cheers, I forgot to initialize it in bodygraph.h